conda_build/metadata.py

```diff
@@ -279,15 +279,18 @@ def check_fields(meta, strictness=.9):
     try:
         _ = _meta['package']['name']
         _ = _meta['package']['version']
-    except IndexError:
-        raise ValueError("Invalid recipe")
+    except KeyError:
+        sys.exit("Invalid meta.yaml: Missing package/name or package/version.")
 
     for mk in meta.get_all_keys():
         close_matches = set(get_close_matches(mk, RECOGNIZED_FIELDS, cutoff=strictness))
 
-        # if we don't have an exact match, notify the user
-        if mk not in close_matches:
-            print("{} is not a recognized key.  Perhaps you meant {}?".format(mk, ', '.join(close_matches)))
+        # If there was nothing even close, we allow it -- it's a user-defined key.
+        # But if we found a close match, notify the user -- it's probably a typo.
+        if close_matches and mk not in close_matches:
+            sys.exit("In meta.yaml: {} is not a recognized key, but it's similar to a standard key.\n"
+                     "Perhaps you meant {}?".format(mk, ', '.join(close_matches)))
```

Comment:
Again, no sys.exit.  I like you error message though.  It implicitly encourages users to choose keys that aren't close to standard keys.